### PR TITLE
Update subqueries building import query based on neo4j version

### PR DESCRIPTION
**Pull request by willtai**
# Summary
`import_query` uses Cypher syntax that is only available for Neo4j version 5.23 and above. This PRs adds a check to build the required query string.

# Testing
This has been manually tested for both Neo4j versions 5.22 and 5.23. The `graph_prompting.py` script was used, which is adapted from [this notebook](https://github.com/langchain-ai/langchain/blob/master/docs/docs/how_to/graph_prompting.ipynb). 

Unit tests were added.

### CODE REVIEW

File: libs/neo4j/pyproject.toml [hunk 1]
```diff
@@ -16,6 +16,7 @@ python = ">=3.9,<4.0"
 langchain-core = "^0.3.0"
 neo4j = "^5.25.0"
 langchain-community = "^0.3.3"
+langchain-openai = "^0.2.8"
```
Comment (stellasia) [inline libs/neo4j/pyproject.toml:19]: Should this dependency be optional?

Comment (willtai) [reply libs/neo4j/pyproject.toml]: https://github.com/langchain-ai/langchain-neo4j/pull/10/commits/e3c7e2504e869d815ffa3ac3215d06a241d4756c

File: libs/neo4j/examples/graph_prompting.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
+import os
```
[the remark this answers is not in the document]
Comment (willtai) [reply libs/neo4j/examples/graph_prompting.py]: https://github.com/langchain-ai/langchain-neo4j/pull/10/commits/e3c7e2504e869d815ffa3ac3215d06a241d4756c

File: libs/neo4j/langchain_neo4j/vectorstores/neo4j_vector.py [hunk 3]
```diff
@@ -928,6 +918,32 @@ def add_embeddings(
...
         return ids
 
+    def _build_import_query(self) -> str:
+        """
+        Build the Cypher import query string based on the Neo4j version.
+
+        Returns:
+            str: The constructed Cypher query string.
+        """
+        if self.neo4j_version_is_5_23_or_above:
```
Comment (oskarhane) [inline libs/neo4j/langchain_neo4j/vectorstores/neo4j_vector.py:928]: Where is this being set?

File: libs/neo4j/pyproject.toml [hunk 4]
```diff
@@ -27,6 +27,12 @@ pytest-socket = "^0.7.0"
 pytest-watcher = "^0.3.4"
 langchain-core = {git = "https://github.com/langchain-ai/langchain.git", subdirectory = "libs/core"}
 
+[tool.poetry.group.langchain_openai]
+optional = true
+
+[tool.poetry.group.langchain_openai.dependencies]
+langchain-openai = "^0.2.8"
```
Comment (alexthomas93) [inline libs/neo4j/pyproject.toml:34]: We probably don't need this as an optional dependency if you've removed the example

File: libs/neo4j/langchain_neo4j/vectorstores/neo4j_vector.py (hunk 3 above)
Comment (willtai) [reply libs/neo4j/langchain_neo4j/vectorstores/neo4j_vector.py:928]: https://github.com/langchain-ai/langchain-neo4j/blob/update-deprecated-cypher/libs/neo4j/langchain_neo4j/vectorstores/neo4j_vector.py#L672

Comment (oskarhane) [reply libs/neo4j/langchain_neo4j/vectorstores/neo4j_vector.py:928]: Thanks, wanted to make sure the aura versions work :)

File: libs/neo4j/pyproject.toml (hunk 4 above)
Comment (willtai) [reply libs/neo4j/pyproject.toml]: Removed this

File: libs/neo4j/pyproject.toml [hunk 5]
```diff
@@ -54,6 +54,7 @@ optional = true
 
 [tool.poetry.group.dev.dependencies]
 langchain-core = {git = "https://github.com/langchain-ai/langchain.git", subdirectory = "libs/core"}
+langchain-openai = "^0.2.8"
```
Comment (alexthomas93) [inline libs/neo4j/pyproject.toml:57]: Looks like langchain-openai is still here. Might be worth just dropping any changes to the poetry files for this PR unless you've added any other deps

**Comment by alexthomas93** [comment]
@CodiumAI-Agent /update_changelog

File: CHANGELOG.md [hunk 6]
```diff
@@ -7,3 +7,7 @@
 ### Added
 
 - Migrated all Neo4j-related code, including tests and integration tests, from the LangChain Community package to this package.
+
+### Updated
+- Improved subquery building for import queries to support different Neo4j versions.
+- Added unit tests for Neo4j version-specific query construction.
```
Comment (alexthomas93) [inline CHANGELOG.md:13]:
```suggestion
### Fixed
- Fixed bugs in the Neo4jVector and GraphCypherQAChain classes preventing these classes from working with versions < 5.23 of Neo4j
```
Minor point but I feel like the bot suggestion here was a bit rubbish. Maybe replace with something like this?

Comment (alexthomas93) [reply CHANGELOG.md:13]: Also shouldn't this be in the Next section anyway?

Comment (willtai) [reply CHANGELOG.md]: https://github.com/langchain-ai/langchain-neo4j/pull/10/commits/a4fde35884e5b34702d304812d33ecdb1a0c16be

**Review by alexthomas93 — APPROVED** [review]
LGTM

**Pull request merged**